File: Day 4/tasks/fizzBuzz/fizzBuzz.js
// ? Write a function which plays the game of FizzBuzz
// ? The function accepts a number N and counts from 1 to N in according to the following rules:

// ? If the number is divisible by 3 -> print out 'Fizz'
// ? If the number is divisible by 5 -> print out 'Buzz'
// ? If the number is divisible both by 3 and 5 -> print out 'FizzBuzz'
// ? In any other case -> print out the number

function fizzBuzz(n) {
	for (let i = 1; i <= n; i++) {
		// Firstly we must check the 'FizzBuzz' scenario
		// Because it is a more specific case - covering both divisibility by 3 AND 5
		if (i % 3 === 0 && i % 5 === 0) {
			console.log('FizzBuzz')
			// Then, we look for divisibility by ONLY one number
		} else if (i % 3 === 0) {
			console.log('Fizz')
		} else if (i % 5 === 0) {
			console.log('Buzz')
		} else {
			console.log(i)
		}
	}
}

fizzBuzz(15)

// ? Can this be improved somehow? Possibly shortened?

File: Day 4/algorithms/practice/fizzBuzz.py
# Write a function which plays the game FizzBuzz
# This function will accept a number N => 1 to N

# If the number is divisible by 3 -> print out "Fuzz" on a new line
# If the number is divisible by 5 -> print out "Buzz" on a new line
# If the number is divisible by 7 -> print out "Bizz" on a new line
# If the number is divisible by BOTH 3 and 5 -> print out "FuzzBuzz" on a new line
# If the number is divisible by BOTH 3 and 5 and 7 -> print out "FuzzBuzzBizz" on a new line
# In any other case -> print out the actual number

def fizz_buzz(n):
    # Loop from 1 to N
    for i in range(1, n+1):
        # First check the more SPECIFIC scenario
        if i % 3 == 0 and i % 5 == 0 and i % 7 == 0:
            print("FizzBuzzBizz")
        if i % 3 == 0 and i % 5 == 0:
            print("FizzBuzz")
        elif i % 3 == 0:
            print("Fizz")
        elif i % 5 == 0:
            print("Buzz")
        elif i % 7 == 0:
            print("Bizz")
        else:
            print(i)


fizz_buzz(15)
